reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/TakeWhilePredicate.kt

```diff
@@ -0,0 +1,33 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.base.tryCatch
+import com.badoo.reaktive.completable.CompletableCallbacks
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.disposable.DisposableWrapper
+
+/**
+ * Emit items by upstream as long as `predicate` is satisfied,
+ * and then completes as soon as this condition is not satisfied.
+ */
+fun <T> Observable<T>.takeWhile(predicate: (T) -> Boolean): Observable<T> =
+    observable { emitter ->
+        val disposableWrapper = DisposableWrapper()
+        emitter.setDisposable(disposableWrapper)
+
+        subscribe(
+            object : ObservableObserver<T>, CompletableCallbacks by emitter {
+                override fun onSubscribe(disposable: Disposable) {
+                    disposableWrapper.set(disposable)
```

Comment:
No need for `DisposableWrapper` here, just call `emitter.set(disposable)`, check e.g. `map` operator. Would be nice to update `takeUntil` in the same way.

Reply:
Done